tweetpic/function/handler.py

```diff
@@ -49,12 +70,29 @@ def handle(req):
         im.save(filename, "JPEG")
         image = open(filename, 'rb')
 
-        status = api.PostUpdate("I colourised your image in %.1f seconds. Find out more: https://subr.pw/s/cmpb5x7" % duration,
-            media=image,
-            auto_populate_reply_metadata=True,
-            in_reply_to_status_id=in_reply_to_status_id)
+        try:
+            status = api.PostUpdate("I colourised your image in %.1f seconds. Find out how: https://subr.pw/s/cmpb5x7" % duration,
+                media=image,
+                auto_populate_reply_metadata=True,
+                in_reply_to_status_id=in_reply_to_status_id)
+
+            return {
+                "reply_to": in_reply_to_status_id,
+                "status_id": status.id
+            }
+
+        except twitter.error.TwitterError, e:
+            for m in e.message:
+                if m['code'] == 34 or m['code'] == 385:
+                    print('Tweet %i went missing' % in_reply_to_status_id)
+                    break
+                if m['code'] == 88:
+                    print('We hit the API limits, queuing %i' % in_reply_to_status_id)
+                    requeue(req)
+                    break
+
         image.close()
         return {
             "reply_to": in_reply_to_status_id,
```

Comment:
What happens here if we hit the requeue block?

Reply:
If we got an error, it will loop through all the error messages `e.message[]`. If any of those errors are `385` (API limit) then we requeue. The `break` was to get out of the error check loop but we probably don't care about it, seeing as we won't have multiple instances of the same error. I'll add a `finally` statement which will run after getting an error.